File: skykeys/netlify/functions/core/normalize.ts
/**
 * Offer normalization helpers for Netlify Functions.
 *
 * Translate raw adapter payloads into the canonical SkyKeys offer schema,
 * including currency conversion, fee placeholders, and metadata flags.
 */

// TODO: Define TypeScript interfaces for raw adapter output and normalized
// offers to enforce shape consistency across the pipeline.

// TODO: Port normalization steps from the Python scaffolding, ensuring
// price_total_eur = price_eur + fees_eur and flags.fees_estimated semantics are
// preserved.

File: skykeys/backend/core/normalize.py
"""Normalization utilities for SkyKeys offer data.

Each function should accept raw scraped payloads and emit normalized
structures aligned with the canonical schema (`title`, `shop`,
`shop_domain`, `platform`, `region`, `url`, `base_price`, `currency`,
`price_eur`, `fees_eur`, `price_total_eur`, `source`, `scraped_at`,
`flags`).

Implementation guidelines:
- Currency conversion must rely on :mod:`skykeys.backend.core.currency`.
- Canonical title derivation must strip stopwords and clamp to 24 chars.
- Flags should capture data quality issues and estimation hints.
"""

from __future__ import annotations

# TODO: define dataclasses or TypedDicts representing normalized offers
# while keeping the module purely declarative for now.


def normalize_offer(raw_offer: dict) -> dict:
    """Placeholder for the normalization pipeline.

    TODO: implement transformation, invoking helper functions for
    currency conversion, canonical title generation, platform/region
    fallback resolution, and fee estimation metadata.
    """

    raise NotImplementedError("Normalization logic pending implementation")


# TODO: add helper functions such as `canonicalize_title`,
# `infer_platform`, and `ensure_currency` with comprehensive docstrings
# and unit-test coverage.
